cmd/eksctl-anywhere/cmd/upgradeplancluster.go

```diff
@@ -89,13 +91,28 @@ func (uc *upgradeClusterOptions) upgradePlanCluster(ctx context.Context) error {
 		return err
 	}
 
-	componentChangeDiffs := eksaupgrader.EksaChangeDiff(currentSpec, newClusterSpec)
-	if componentChangeDiffs == nil {
-		componentChangeDiffs = &types.ChangeDiff{}
+	componentChangeDiffs := &types.ChangeDiff{}
+
+	if newClusterSpec.Cluster.IsSelfManaged() {
+		client, err := deps.UnAuthKubeClient.BuildClientFromKubeconfig(managementCluster.KubeconfigFile)
+		if err != nil {
+			return err
+		}
+
+		currentManagementComponentsVersionsBundle, err := cluster.GetManagementComponentsVersionsBundle(ctx, client, currentSpec.Cluster)
+		if err != nil {
+			return err
+		}
+
+		newVersionsBundle := &newClusterSpec.Bundles.Spec.VersionsBundles[0]
+
+		componentChangeDiffs.Append(eksaupgrader.EksaChangeDiff(currentManagementComponentsVersionsBundle, newVersionsBundle))
```

Comment:
I think we can reduce the change to *upgrader's changeDiff methods' interfaces. As you can get currentManagementComponentsVersionsBundle and newVersionsBundle inside each changeDiff method.

Reply:
That's true, but @g-gaston and I had discussed the details of this earlier. We landed upon keeping the fetching logic separate from business logic to maintain the level of modularity. The tradeoff is the interface change on the change diff methods.